Add Value * == Value * peephole

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #56966 [JIT] Disable Complete Shape Inlining For Testing Purposes
* #56828 Add unary/binary ops necessary for mobilenet
* #56642 Only optimize while change has been made
* #56438 [JIT] integer value refinement
* #56437 Factor out isDominatedBy
* **#55978 Add Value * == Value * peephole**
* #55977 [JIT] [Easy] Factor out non tensor peephole
* #55926 Add list len refinement
* #55925 Add handling of symbolic shapes
* #54809 Initial Symbolic Shape Analysis

This is needed for broadcasting two of the same symbolic shape

Differential Revision: [D27755328](https://our.internmc.facebook.com/intern/diff/D27755328)

Commit:
Add Value * == Value * peephole

[ghstack-poisoned]

Commit:
Update on "Add Value * == Value * peephole"


This is needed for broadcasting two of the same symbolic shape

Differential Revision: [D27755328](https://our.internmc.facebook.com/intern/diff/D27755328)

[ghstack-poisoned]

Commit:
Update on "Add Value * == Value * peephole"


This is needed for broadcasting two of the same symbolic shape

Differential Revision: [D27755328](https://our.internmc.facebook.com/intern/diff/D27755328)

[ghstack-poisoned]

Commit:
Update on "Add Value * == Value * peephole"


This is needed for broadcasting two of the same symbolic shape

Differential Revision: [D27755328](https://our.internmc.facebook.com/intern/diff/D27755328)

[ghstack-poisoned]

Commit:
Update on "Add Value * == Value * peephole"


This is needed for broadcasting two of the same symbolic shape

Differential Revision: [D27755328](https://our.internmc.facebook.com/intern/diff/D27755328)

[ghstack-poisoned]

Commit:
Update on "Add Value * == Value * peephole"


This is needed for broadcasting two of the same symbolic shape

Differential Revision: [D27755328](https://our.internmc.facebook.com/intern/diff/D27755328)

[ghstack-poisoned]

Commit:
Update on "Add Value * == Value * peephole"


This is needed for broadcasting two of the same symbolic shape

Differential Revision: [D27755328](https://our.internmc.facebook.com/intern/diff/D27755328)

[ghstack-poisoned]

Commit:
Update on "Add Value * == Value * peephole"


This is needed for broadcasting two of the same symbolic shape

Differential Revision: [D27755328](https://our.internmc.facebook.com/intern/diff/D27755328)

[ghstack-poisoned]

penguinwu commented on torch/csrc/jit/passes/peephole_non_tensor.cpp:
```diff
@@ -114,6 +114,26 @@ struct PeepholeOptimizeNonTensorImpl {
+        switch (inp_kind) {
+          case TypeKind::BoolType:
+          case TypeKind::IntType:
+          case TypeKind::FloatType: {
```
What about `ComplexType` and `StringType`?

BTW, in which case would `x == x` throw exceptions on mutable types?

eellison replied:
Tensor and anything containing Tensor

penguinwu reviewed (approved):
LGTM, just one minor question

Commit:
Update on "Add Value * == Value * peephole"


This is needed for broadcasting two of the same symbolic shape

Differential Revision: [D27755328](https://our.internmc.facebook.com/intern/diff/D27755328)

[ghstack-poisoned]

eellison committed:
Update on "Add Value * == Value * peephole"


This is needed for broadcasting two of the same symbolic shape

Differential Revision: [D27755328](https://our.internmc.facebook.com/intern/diff/D27755328)

[ghstack-poisoned]

eellison committed:
Update on "Add Value * == Value * peephole"


This is needed for broadcasting two of the same symbolic shape

Differential Revision: [D27755328](https://our.internmc.facebook.com/intern/diff/D27755328)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D27755328).

eellison committed:
Update on "Add Value * == Value * peephole"


This is needed for broadcasting two of the same symbolic shape

Differential Revision: [D27755328](https://our.internmc.facebook.com/intern/diff/D27755328)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D27755328).

eellison committed:
Update on "Add Value * == Value * peephole"


This is needed for broadcasting two of the same symbolic shape

Differential Revision: [D27755328](https://our.internmc.facebook.com/intern/diff/D27755328)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D27755328).

eellison committed:
Update on "Add Value * == Value * peephole"


This is needed for broadcasting two of the same symbolic shape

Differential Revision: [D27755328](https://our.internmc.facebook.com/intern/diff/D27755328)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D27755328).

eellison committed:
Update on "Add Value * == Value * peephole"


This is needed for broadcasting two of the same symbolic shape

Differential Revision: [D27755328](https://our.internmc.facebook.com/intern/diff/D27755328)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D27755328).

eellison committed:
Update on "Add Value * == Value * peephole"


This is needed for broadcasting two of the same symbolic shape

Differential Revision: [D27755328](https://our.internmc.facebook.com/intern/diff/D27755328)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D27755328).
